homeassistant/helpers/translation.py

```diff
@@ -311,4 +310,7 @@ async def async_get_translations(
             cache = hass.data[TRANSLATION_FLATTEN_CACHE] = _TranslationCache(hass)
         cached = await cache.async_fetch(language, category, components)
 
-    return dict(ChainMap(*cached))
+    result = {}
+    while cached:
+        result.update(cached.pop())
```

Comment:
Why even pop and not just iterate over it ?

Reply:
Good idea. Memory isn't a concern here